descartes_light/solvers/include/descartes_light/solvers/bgl/dfs_add_all_solver.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2021, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef DESCARTES_LIGHT_DFS_ALL_SOLVER_H
+#define DESCARTES_LIGHT_DFS_ALL_SOLVER_H
+
+#include <descartes_light/solvers/bgl/bgl_solver.h>
+
+namespace descartes_light
+{
+/**
+ * @brief Depth First Search that adds edged from source node to all ndes in the target rung
+ */
+template <typename FloatType>
+class DFSAddAllSolver : public BGLSolverBaseSVDE<FloatType>
```

Comment:
I would consider renaming to something like `DFSSolverSVDE` or `DepthFirstSVDESolver` to be more descriptive than just `AddAll`